### DISCUSSION TITLE
python warning message for METviewer batch job

### DISCUSSION THREAD

**Discussion by Ho-ChunHuang-NOAA** [dtcenter/METplus#1988] [Plot Generation] [answered]
When running the metviewer batch job from WCOSS2, receiving warning message as

Timeseries
==== Start Python error  ====
/opt/metviewer_python/METcalcpy/metcalcpy/util/utils.py:707: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  output_ee_data = output_ee_data.append(series_data_after_ee)
/opt/metviewer_python/METcalcpy//metcalcpy/sum_stat.py:238: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  aggregated_values = aggregated_values.append(point_data)
:::::::::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::;;
/opt/metviewer_python/METcalcpy//metcalcpy/sum_stat.py:238: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  aggregated_values = aggregated_values.append(point_data)
====   End Python error  ====

Taylor Diagram
==== Start Python error  ====
/opt/metviewer_python/METcalcpy/metcalcpy/util/utils.py:707: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  output_ee_data = output_ee_data.append(series_data_after_ee)
:::::::::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::;;
/opt/metviewer_python/METcalcpy/metcalcpy/util/utils.py:707: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  output_ee_data = output_ee_data.append(series_data_after_ee)
====   End Python error  ====
Performance plot seems to be free of similar message.

Does user need to update the script on their end? and if yes, please advise.

**Comment by JohnHalleyGotway** [comment]
Ho-Chun, these `FutureWarning` messages you're seeing when running the METviewer batch jobs can safely be ignored for now.

Functionality in python changes over time. Prior to removing/replacing functionality, python prints the `FutureWarning` messages to let users know what will be deprecated in future versions. So yes, the METplus developers should pay careful attention to these and modify the code to stop using python functionality that will change. In this case, we need to switch from `pandas.append` to `pandas.concat`.

In general, users do need to worry about these types of warnings. But thanks for bringing this particular one to our attention.

@TatianaBurek and/or @bikegeek, can you please write up an issue for the correct METplus analysis tools component to replace `append` with `concat` throughout?

**Comment by bikegeek** [comment]
Hello Ho-Chun, we have created a Github issue to fix this deprecation that arises in the pandas third-party package:

https://github.com/dtcenter/METcalcpy/issues/249

**Comment by Ho-ChunHuang-NOAA** [comment, marked as answer]
Thanks.  I understand it will be fixed in the future and will not impact
current operation.

Please close this issue.

Ho-Chun Huang, Ph.D.

**Reply by bikegeek** [reply]
Closing this discussion.  Please create a new discussion if you need to add anything, or add a comment to the Github Issue: https://github.com/dtcenter/METcalcpy/issues/249
